verifier/src/main/kotlin/tech/figure/classification/asset/verifier/util/eventstream/providers/BlockApiEventStreamProvider.kt

```diff
@@ -45,17 +45,16 @@ class BlockApiEventStreamProvider(
         try {
             while (coroutineScope.isActive) {
                 (from..current).forEach { blockHeight ->
-                    if (from > current) return@forEach
                     process(blockHeight, onBlock, onEvent, onError)
-                    lastProcessed.set(blockHeight)
+                    lastProcessed.set(blockHeight + 1)
                 }
 
                 // We've reached the current block, so fire the completion event
                 onCompletion(null)
 
                 // Once we've met the current block, no need to keep spinning. Wait here for 4 seconds and process again.
                 delay(DEFAULT_BLOCK_DELAY_MS.milliseconds)
-                from = lastProcessed.incrementAndGet()
+                from = lastProcessed.get()
```

Comment:
I think this will probably do the trick, but if it ends up not doing that, you might want to add an inner loop here (rip) that continuously delays until `current > lastProcessed.get()`, and  put the block height set in `lastProcessed` back to what it is now.  Then you'd be guaranteed to only start a new iteration with unprocessed blocks